pkg/kubelet/userns_manager_test.go

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/types"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	pkgfeatures "k8s.io/kubernetes/pkg/features"
+)
+
+type testUserNsPodsManager struct {
+}
+
+func (m *testUserNsPodsManager) getPodDir(podUID types.UID) string {
+	return "/tmp/non-existant-dir.This-is-not-used-in-tests"
+}
+
+func (m *testUserNsPodsManager) listPodsFromDisk() ([]types.UID, error) {
+	return nil, nil
+}
+
+func TestUserNsManagerAllocate(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.UserNamespacesSupport, true)()
+
+	testUserNsPodsManager := &testUserNsPodsManager{}
+	m, err := MakeUserNsManager(testUserNsPodsManager)
+	assert.NoError(t, err)
```

Comment:
```suggestion
	require.NoError(t, err)
```

prefer require over assert, assert will not stop the test from executing if the condition fails, require will.

Otherwise you need to handle when `m` is nil, because the resulting panics are always harder to triage.